### ISSUE TITLE
Package Blade directives no longer working since 5.x

### ISSUE THREAD

**Issue by engel-m** [Laravel-Backpack/CRUD#4168]
I followed the upgrade guide of Backpack 5.x, and this caused all blade directives from packages other than Backpack to fail and just parse on the blade view as text instead of doing something. And notably this happens OUTSIDE of the Backpack admin panel environment.

Examples of failing Blade directives that just get parsed on the page:

@impersonating, @endImpersonating, (Lab404/Impersonate for Laravel)
@hasanyrole, @endhasanyrole (Spatie Permissions for Laravel)

What can have caused this? Somehow because of the Backpack upgrade they don't get recognised anymore.

**Comment by pxpm** [comment]
Hello @engel-m 

I am sorry but I am unable to reproduce your issue. 

I am using one of the packages you refered, spatie permissions, I use in blade file:
```blade
@hasrole('superadmin', 'backpack')
<h1> Hello superadmin! </h1>
@endhasrole
```

![image](https://user-images.githubusercontent.com/7188159/153223335-e20a0642-baf0-4c0f-893c-0ea9d8d2ede4.png)

As you can see in the package https://github.com/DigitallyHappy/assets, we just register more custom blade directives, like those packages you mentioned do, for different purposes.

I am unsure on what migth be going on, but you could try `composer dump-autoload` and `php artisan cache:clear` ? 

Let me know how it goes,
Pedro

**Comment by tabacitu** [comment]
In addition to what @pxpm said above, I would try a `php artisan view:clear` too. Very weird @engel-m - we'll help you out as much as we can, but I'm pretty sure there's nothing Backpack can do to stop those from working. 

Did you do anything else at the same time as upgrading Backpack? Did you perhaps upgrade to Laravel 9 too?

**Comment by engel-m** [comment]
@tabacitu @pxpm I did clear all my cache/view/config and even tried manually removing the bootstrap/cache dir. Also tried republishing package assets for the packages that stopped working. But nothing seems to work. 

The only thing that works for me, is a downgrade to Backpack 4.1

I did not install anything together with the Backpack upgrade.

**Comment by engel-m** [comment]
Yes I do have everything set up normally when it comes to Blade files. The rest of the views work fine and all other assets are loaded. It's only that somehow these directives are not interpreted when rendering the views. The packages themselves do seem to work.

I am completely in the dark trying to find out why it is not working.

**Comment by tabacitu** [comment]
Ok so I went quite deep into this rabbit hole 😅 But I think (I hope!) I found something 🎉🎉🎉

Apparently there might be an issue when packages load blade directives using `Blade::directive()` like apps do. But not for everybody, and not all packages 😡🤯! That's why some packages - like Impersonate and Laravel-Permission have resorted to loading their directives not using the above, but using Laravel's `afterResolve()`. Which should prevent one package from interfering or triggering the load for other packages.

**Long story short - I've made a PR to `digitallyhappy/assets` to load blade directives the same way these two packages do.** Since they're such popular packages, even if we don't agree that's a better way to do it (but we do), we'd still have to do it _their way_ in order to provide compatibility between Backpack and them.

@engel-m can you do me a favor and try out this branch - https://github.com/DigitallyHappy/assets/pull/11#issuecomment-1034579257 and tell us if that solves the problem for you? If it does, we'll merge it into `main` and launch it as `2.0.3`.

@pxpm could you please try it too, and tell me if these other blade directives _keep working_ for you?

If it works for us all, I think it's safe to merge.
Thanks!

**Comment by engel-m** [comment]
@tabacitu Thank you so much for putting in the effort to investigate this! This PR seems to fix the problem with both packages / blade directives, as far as I can see.
